index.js

```diff
@@ -122,115 +152,34 @@ const defaultPlugins = {
-
-    // globals, aka replace require calls with this
-    globals: options.globals ? options.globals(Object.assign({}, defaultGlobals)) : defaultGlobals,
-
-    // plugin order by build type
-    plugins: options.plugins ? options.plugins(Object.assign({}, defaultPlugins)) : defaultPlugins,
-
-    // banner to add to the top of all bundles
-    banner: options.banner || `/*! @name ${pkg.name} @version ${pkg.version} @license ${pkg.license} */`,
-
-    // ignore tests, external modules, and package.json
-    excludeCoverage: ['test/**', path.join(__dirname, '**'), 'node_modules/**', 'package.json'],
-
-    // if we should include istanbul by default
-    coverage: typeof options.coverage === 'boolean' ? options.coverage : true
-  };
-
-  if (!options.coverage) {
-    defaultPlugins.test.splice(defaultPlugins.test.indexOf('istanbul'), 1);
-  }
-
-  const defaultBabel = () => {
-    return {
-      babelrc: false,
-      exclude: path.join(process.cwd(), 'node_modules/**'),
-      presets: [
-        [presetEnv, {loose: true, modules: false, targets: {browsers: settings.browserslist}}]
-      ],
-      plugins: [
-        transformObjectAssign
-      ]
-    };
-  };
-
-  // get babel settings from the users provided function or use the defaults
-  settings.babel = options.babel ? options.babel(defaultBabel()) : defaultBabel();
-
-  if (options.excludeCoverage) {
-    settings.excludeCoverage = options.excludeCoverage(settings.excludeCoverage);
-  }
-
-  // primed plugins
-  settings.primedPlugins = {
-    babel: babel(settings.babel),
-    commonjs: commonjs({sourceMap: false}),
-    json: json(),
-    multiEntry: multiEntry({exports: false}),
-    resolve: resolve({browser: true, mainFields: ['module', 'jsnext', 'main']}),
-    uglify: terser({output: {comments: 'some'}, include: [/^.+\.min\.js$/]}),
-    istanbul: istanbul({exclude: settings.excludeCoverage})
-  };
-
-  if (options.primedPlugins) {
-    settings.primedPlugins = options.primedPlugins(settings.primedPlugins);
-  }
-
-  Object.keys(settings.plugins).forEach((buildType) => {
-
-    // verify that listed plugins are an object or
-    const verifyPlugins = (pluginName) => {
-      if (typeof pluginName !== 'string') {
-        return;
-      }
-
-      if (!settings.primedPlugins[pluginName]) {
-        throw new Error(`Plugin ${pluginName} listed in ${buildType} does not have an equivent Primed Plugin`);
-      }
-    };
+  const basicName = pkg.name.split('/').reverse()[0];
+  const settings = Object.assign({
+    input: 'src/plugin.js',
+    testInput: 'test/**/*.test.js',
+    distName: basicName,
+    exportName: basicName.replace(/-(\w)/g, (matches, letter) => letter.toUpperCase()),
+    browserslist: pkg.browserslist || ['defaults', 'ie 11'],
+    checkWatch: true,
+    banner: `/*! @name ${pkg.name} @version ${pkg.version} @license ${pkg.license} */`,
+    coverage: true
+  }, options || {});
+
+  Object.keys(ORDERED_DEFAULTS).forEach(function(key) {
```

Comment:
Since there are ordered defaults we can loop through them rather than doing the same thing over an over with each option that can be a function with default value as the first arg.